packages/connections/src/utils/validation-errors.spec.ts

```diff
@@ -0,0 +1,284 @@
+      expect(result).to.deep.equal([
+        {
+          message: 'TLS must be enabled in order to use x509 authentication.',
+        },
+      ]);
+    });
+    it('should return error if tls is disabled', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://myserver.com?authMechanism=MONGODB-X509&tls=false&tlsCertificateFile=/path/to/file.pem`,
+        },
+      };
+      const result = validateConnectionInfoErrors(connectionInfo);
+      expect(result).to.deep.equal([
+        {
+          message: 'TLS must be enabled in order to use x509 authentication.',
+        },
+      ]);
+    });
+
+    it('should return error if ssl is disabled', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://myserver.com?authMechanism=MONGODB-X509&ssl=false&tlsCertificateFile=/path/to/file.pem`,
+        },
+      };
+      const result = validateConnectionInfoErrors(connectionInfo);
+      expect(result).to.deep.equal([
+        {
+          message: 'TLS must be enabled in order to use x509 authentication.',
+        },
+      ]);
+    });
+    it('should not return error if tls is enabled', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://myserver.com?authMechanism=MONGODB-X509&tls=true&tlsCertificateFile=/path/to/file.pem`,
+        },
+      };
+      const result = validateConnectionInfoErrors(connectionInfo);
+      expect(result).to.be.empty;
+    });
+
+    it('should not return error if ssl is enabled', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://myserver.com?authMechanism=MONGODB-X509&ssl=true&tlsCertificateFile=/path/to/file.pem`,
+        },
+      };
+      const result = validateConnectionInfoErrors(connectionInfo);
+      expect(result).to.be.empty;
+    });
+
+    it('should return error if no certificate is provided', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://myserver.com?authMechanism=MONGODB-X509&ssl=true`,
+        },
+      };
+      const result = validateConnectionInfoErrors(connectionInfo);
+      expect(result).to.deep.equal([
+        {
+          message: 'A Client Certificate is required with x509 authentication.',
+        },
+      ]);
+    });
+  });
+  describe('LDAP', function () {
+    it('should return errors if no username is defined', function () {
+      const connectionInfo: ConnectionInfo = {
+        id: 'connection-test',
+        connectionOptions: {
+          connectionString: `mongodb://:password@myserver.com?authMechanism=PLAIN`,
```

Comment:
i think this is going to break before getting here as for the other similar cases we could just assume this case is handled somewhere else